test/boost/token_metadata_test.cc

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <boost/test/unit_test.hpp>
+#include <seastar/testing/thread_test_case.hh>
+#include "utils/fb_utilities.hh"
+#include "locator/token_metadata.hh"
+#include "locator/simple_strategy.hh"
+
+using namespace locator;
+
+namespace {
+    const auto ks_name = sstring("test-ks");
+
+    endpoint_dc_rack get_dc_rack(inet_address) {
+        return {
+            .dc = "unk-dc",
+            .rack = "unk-rack"
+        };
+    }
+
+    token_metadata create_token_metadata(inet_address this_endpoint) {
+        return token_metadata({
+            topology::config {
+                .this_host_id = host_id::create_random_id(),
+                .this_endpoint = this_endpoint,
+                .local_dc_rack = get_dc_rack(this_endpoint)
+            }
+        });
+    }
+}
+
+SEASTAR_THREAD_TEST_CASE(test_pending_endpoints_for_bootstrap_first_node) {
```

Comment:
I don't like unit test files with ~200 lines of code. We'll compile hundreds of thousands of headers file lines just to emit these.

No need to change. I propose we do a consolidation pass later where we merge related small files and split overgrown large files.

Reply:
There's locator_topology_test, which is a candidate for unification with this.

Reply:
OTOH, I'm not sure - does this test enough this code?

Reply:
> There's locator_topology_test, which is a candidate for unification with this.

I'll create a follow-up PR with the merge.

Reply:
> OTOH, I'm not sure - does this test enough this code?

These tests cover only aspects of token_metadata important for the refactoring, not the entire scope.